From 65da221b278c9975e97cd9efef253b5d332a2e8f Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Fri, 7 Feb 2020 20:50:43 +0530
Subject: [PATCH] LSM: Update xfrm_policy_lookup prototype to fix build error

Kernel build using clang fails with incompatible pointer types.

As passed secids is not modified, changed xfrm_policy_lookup
prototype accordingly.

Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 include/linux/lsm_hooks.h | 2 +-
 include/linux/security.h  | 4 ++--
 security/security.c       | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/linux/lsm_hooks.h b/include/linux/lsm_hooks.h
index 381dae9c3ec5..5ef2d5111151 100644
--- a/include/linux/lsm_hooks.h
+++ b/include/linux/lsm_hooks.h
@@ -1745,7 +1745,7 @@ union security_list_options {
 	void (*xfrm_state_free_security)(struct xfrm_state *x);
 	int (*xfrm_state_delete_security)(struct xfrm_state *x);
 	int (*xfrm_policy_lookup)(struct xfrm_sec_ctx *ctx,
-					struct secids *fl_secid, u8 dir);
+					const struct secids *fl_secid, u8 dir);
 	int (*xfrm_state_pol_flow_match)(struct xfrm_state *x,
 						struct xfrm_policy *xp,
 						const struct flowi *fl);
diff --git a/include/linux/security.h b/include/linux/security.h
index e98b45ab7a1e..f9d406d9695d 100644
--- a/include/linux/security.h
+++ b/include/linux/security.h
@@ -1597,7 +1597,7 @@ int security_xfrm_state_alloc_acquire(struct xfrm_state *x,
 int security_xfrm_state_delete(struct xfrm_state *x);
 void security_xfrm_state_free(struct xfrm_state *x);
 int security_xfrm_policy_lookup(struct xfrm_sec_ctx *ctx,
-					struct secids *fl_secid, u8 dir);
+					const struct secids *fl_secid, u8 dir);
 int security_xfrm_state_pol_flow_match(struct xfrm_state *x,
 				       struct xfrm_policy *xp,
 				       const struct flowi *fl);
@@ -1650,7 +1650,7 @@ static inline int security_xfrm_state_delete(struct xfrm_state *x)
 }
 
 static inline int security_xfrm_policy_lookup(struct xfrm_sec_ctx *ctx,
-					struct secids *fl_secid, u8 dir)
+					const struct secids *fl_secid, u8 dir)
 {
 	return 0;
 }
diff --git a/security/security.c b/security/security.c
index 3b5c6c435f0c..801b7139e249 100644
--- a/security/security.c
+++ b/security/security.c
@@ -2500,7 +2500,7 @@ void security_xfrm_state_free(struct xfrm_state *x)
 }
 
 int security_xfrm_policy_lookup(struct xfrm_sec_ctx *ctx,
-					struct secids *fl_secid, u8 dir)
+					const struct secids *fl_secid, u8 dir)
 {
 	return call_int_hook(xfrm_policy_lookup, 0, ctx, fl_secid, dir);
 }
-- 
2.17.1

